src/frontend/js/pages/DashboardCourses/useOrdersEnrollments.tsx

```diff
@@ -1,28 +1,13 @@
-import { useEffect, useRef, useState } from 'react';
-import { QueryKey, useQueryClient } from '@tanstack/react-query';
-import { defineMessages, useIntl } from 'react-intl';
+import { defineMessages } from 'react-intl';
 import { useJoanieApi } from 'contexts/JoanieApiContext';
-import { Resource } from 'hooks/useResources';
-import {
-  Enrollment,
-  EnrollmentsQuery,
-  Order,
-  PaginatedResourceQuery,
-  PaginatedResponse,
-} from 'types/Joanie';
-import { Maybe } from 'types/utils';
-import { OrderResourcesQuery } from 'hooks/useOrders';
-import { REACT_QUERY_SETTINGS } from 'settings';
-import { PAGE_SIZE } from './constants';
+import { Enrollment, Order, PaginatedResourceQuery } from 'types/Joanie';
+import useUnionResource from 'hooks/useUnionResource';
 
-export enum DataType {
-  ENROLLMENT = 'enrollment',
-  ORDER = 'order',
-}
-
-export type Data = {
-  type: DataType;
-  item: Enrollment | Order;
+export const isOrder = (obj: Order | Enrollment): obj is Order => {
```

Comment:
I get what you wanted to do by removing the `type` but I find that using attributes to determine the type is misleading and is more bug-prone for the future maintenance. As of now if the backend changes an attribute it must keep in mind that the frontend is detecting the type based on this, IMO it breaks the I/O principle ( and SRP ). WDYT all ?